src/components/Rocket.js

```diff
@@ -0,0 +1,54 @@
+/* eslint-disable camelcase */
+import React from 'react';
+import PropTypes from 'prop-types';
+import { useDispatch } from 'react-redux';
+import { reserveRocket, cancelReservation } from '../Redux/Rocket/rockets';
+import './rocket.css';
+
+const Rocket = ({ rocket }) => {
+  const {
+    rocket_id, rocket_name, description, flickr_images, reserved,
+  } = rocket;
+  const dispatch = useDispatch();
+  return (
+    <div className="rocket-detail">
+      <div className="detail-image">
+        <img src={flickr_images} alt={rocket_name} />
+      </div>
+      <div className="rocket-info">
+        <h2 className="rocket-name">{rocket_name}</h2>
+        <p className="rocket-description">
+          <span className={reserved ? 'status' : ' '}>
+            {reserved ? 'Reserved' : ''}
+          </span>
+          {description}
+        </p>
+        <div>
+          <button
+            type="button"
+            className={
+              reserved === true ? 'disabled' : 'reserve-btn'
+            }
+            onClick={() => (reserved
+              ? dispatch(cancelReservation(rocket_id))
+              : dispatch(reserveRocket(rocket_id)))}
+          >
+            {reserved ? 'Cancel Reservation' : 'Reserve Rocket'}
```

Comment:
- [ ] [optional] Please use React conditional rendering syntax for conditions instead of the Ternary operator as this is part of the requirements.
   - You can use the syntax:
   -  `{ reserved && (  // code for reserved rockets )}`
   - `{ !reserved && (  // code for not reserved rockets )}`